db/src/tx_observer.rs

```diff
@@ -155,27 +155,31 @@ impl Drop for TxObservationService {
 }
 
 pub struct InProgressObserverTransactWatcher {
-    collected_attributes: AttributeSet,
     pub txes: IndexMap<Entid, AttributeSet>,
 }
 
 impl InProgressObserverTransactWatcher {
     pub fn new() -> InProgressObserverTransactWatcher {
         InProgressObserverTransactWatcher {
-            collected_attributes: Default::default(),
             txes: Default::default(),
         }
     }
 }
 
 impl TransactWatcher for InProgressObserverTransactWatcher {
+    fn start(&mut self, t: &Entid) {
+        self.txes.entry(t.clone()).or_insert(Default::default());
```

Comment:
Just store the current `txid` in an attribute. Or, simpler, keep the `t` argument to `done`. (This is why I didn't add `begin`, after making those comments: we don't use it yet for any of our watchers.)